ivy/functional/ivy/creation.py

```diff
@@ -1100,6 +1100,158 @@ def eye(
+    out: Optional[ivy.Array] = None,
+) -> ivy.Array:
+    """Returns a array filled with ones on the k diagonal and zeros elsewhere. having
+    the same ``shape`` as an input array ``x``.
+
+
+    Parameters
+    ----------
+    x
+         input array from which to derive the output array shape.
+    k
+        index of the diagonal. A positive value refers to an upper diagonal, a negative
+        value to a lower diagonal, and 0 to the main diagonal. Default: ``0``.
+    dtype
+        output array data type. If dtype is None, the output array data type must be the
+        default floating-point data type. Default: ``None``.
+    device
+        the device on which to place the created array.
+    out
+        optional output array, for writing the result to. It must have a shape that the
+        inputs broadcast to.
+
+    Returns
+    -------
+    ret
+        an array having the same shape as ``x`` and filled with ``ones`` in diagonal ``k``
+        and ``zeros`` elsewhere.
+
+    This function conforms to the `Array API Standard
+    <https://data-apis.org/array-api/latest/>`_. This docstring is an extension of the
+    `docstring <https://data-apis.org/array-api/latest/API_specification/generated/signatures.creation_functions.eye.html>`_ # noqa
+    in the standard.
+
+    Both the description and the type hints above assumes an array input for simplicity,
+    but this func;tion is *nestable*, and therefore also accepts :class:`ivy.Container`
+    instances as a replacement to any of the arguments.
+
+    Functional Examples
+    -------------------
+
+    With :class:`ivy.Array` input:
+
+    >>> x1 = ivy.array([0, 1],[2, 3])
+    >>> y1 = ivy.eye_like(x1)
+    >>> print(y1)
+    ivy.array([[1., 0.],
+               [0., 1.]])
+
+    >>> x1 = ivy.array([0, 1, 2],[3, 4, 5],[6, 7, 8])
+    >>> y1 = ivy.eye_like(x1)
+    >>> print(y1)
+    ivy.array([[1., 0., 0.],
+               [0., 1., 0.],
+               [0., 0., 1.]])
+
+    With :'k' input:
+
+    >>> x1 = ivy.array([0, 1, 2],[3, 4, 5],[6, 7, 8])
+    >>> y1 = ivy.eye_like(x1, k=1)
+    >>> print(y1)
+    ivy.array([[0., 1., 0.],
+               [0., 0., 1.],
+               [0., 0., 0.]])
+
+    With :'out' input:
+
+    >>> x1 = ivy.array([[2, 3, 8],[1, 2, 1],[2, 3, 8],[1, 2, 1]])
+    >>> ivy.eye_like(x1, out=x1)
+    >>> print(x1)
+    ivy.array([[1., 0., 0.],
+               [0., 1., 0.],
+               [0., 0., 1.],
+               [0., 0., 0.]])
+
+    With :class:`ivy.NativeArray` input:
+
+    >>> x1 = ivy.native_array([[3, 8, 2],[2, 8, 3]])
+    >>> y1 = ivy.eye_like(x1)
+    >>> print(y1)
+    ivy.array([[1., 0., 0.],
+               [0., 1., 0.]])
+
+
+    >>> x2 = ivy.native_array([3, 8],[0, 2], [8, 5]])
+    >>> y2 = ivy.eye_like(x2, dtype=ivy.IntDtype('int32'), device=ivy.Device('cpu'))
+    >>> print(y2)
+    ivy.array([[1., 0.],
+               [0., 1.],
+               [0., 0.]])
+
+    # Array ``y2`` is now stored on the CPU.
```

Comment:
These comments should be removed as well

Reply:
done